tensorflow/c/eager/gradient_checker.cc

```diff
@@ -0,0 +1,200 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/c/eager/gradient_checker.h"
+
+#include <memory>
+
+#include "absl/types/span.h"
+#include "tensorflow/c/eager/abstract_tensor_handle.h"
+#include "tensorflow/c/eager/c_api_experimental.h"
+#include "tensorflow/c/eager/c_api_test_util.h"
+#include "tensorflow/c/eager/c_api_unified_experimental.h"
+#include "tensorflow/c/eager/c_api_unified_experimental_internal.h"
+#include "tensorflow/c/eager/gradients.h"
+#include "tensorflow/c/eager/gradients_internal.h"
+#include "tensorflow/c/eager/mnist_gradients_testutil.h"
```

Comment:
You should not be importing testutil in a non-test library.

Reply:
I was using these util functions to create `AbstractTensorHandle`s for inputs to the `Model` such as for getting `thetaPlus` and `thetaMinus`. However, even if I put them in a separate non-`testutil` file, these util functions still have dependencies on other test-only functions for building AbstractTensorHandles on the fly. What would be the best way to resolve this?

Reply:
I believe you are referring to `TestTensorHandleWithDimsFloat` in `c_api_test_util.h`? I am with okay with forking that.

Reply:
Great, that works